modules/monitoring/files/usr/local/bin/govuk_uptime_collector

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env ruby
+
+# Usage: ./collect.rb <environment> <service ...>
+
+require "csv"
+require "fileutils"
+require "net/http"
+require "time"
+require "statsd"
+
+class Array
+  def each_after(n)
+    each_with_index do |elem, i|
+      yield elem if i >= n
+    end
+  end
+end
+
+class Collector
+  def initialize(service, environment)
+    @statsd = Statsd.new("127.0.0.1")
+    @service = service
+    @environment = environment
+  end
+
+  def call
+    while true
+      send_to_statsd(check_status)
+      sleep 5
+    end
+  end
+
+private
+
+  attr_reader :statsd, :service, :environment
+
+  def healthcheck_uri
+    @healthcheck_uri ||= (if environment == "production"
+      URI("https://#{service}.publishing.service.gov.uk/healthcheck")
+    else
+      URI("https://#{service}.#{environment}.publishing.service.gov.uk/healthcheck")
+    end)
```

Comment:
This might be a bit cleaner as:
```
@healthcheck_uri ||= begin 
  prefix = service
  prefix << ".#{environment} unless environment == "production"
  URI("https://#{prefix}.publishing.service.gov.uk/healthcheck")
end
```